src/brigadier_components/string_reader.ts

```diff
@@ -89,6 +94,34 @@ export class StringReader {
         this.cursor += str.length;
         return helper.succeed();
     }
+    /**
+     * Require that a specific string of a set of strings follows
+     * @param opts The options for the expectation
+     */
+    public expectOption(opts: string[]): ReturnedInfo<string> {
+        const helper = new ReturnHelper();
+        let maxlen = 0;
+        const start = this.cursor;
+        for (const opt of opts) {
+            if (opt.length > maxlen) {
+                maxlen = opt.length;
+            }
+            const out = this.expect(opt);
+            if (isSuccessful(out)) {
+                helper.merge(out);
```

Comment:
This `helper.merge()` does nothing.

Personally I think that expectOption should return suggestions. I have refactored ReturnHelper.merge in nbt_cleanup to allow this to be prevented at the merge site if required